tools/moduletests/unit/test_enadiag.py

```diff
@@ -0,0 +1,284 @@
+# Copyright 2016-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""
+Unit tests for the enadiag module
+"""
+import os
+import subprocess
+import sys
+import unittest
+
+import mock
+
+import moduletests.src.enadiag
+
+try:
+    # Python 2.x
+    from cStringIO import StringIO
+except ImportError:
+    # Python 3.x
+    from io import StringIO
+
+if sys.hexversion >= 0x3040000:
+    # contextlib.redirect_stdout was introduced in Python 3.4
+    import contextlib
+else:
+    # contextlib2 is a backport of contextlib from Python 3.5 and is compatible with Python2/3
+    import contextlib2 as contextlib
+
+class Testenadiag(unittest.TestCase):
+    def setUp(self):
+        self.output = StringIO()
+
+    def tearDown(self):
+        self.output.close()
+
+    def test_get_interfaces(self):
+        with mock.patch("moduletests.src.enadiag.os.listdir") as mock_dir:
+            mock_dir.return_value = ["eth0", "eth1", "lo"]
+            self.assertEquals(moduletests.src.enadiag.get_interfaces(),["eth0", "eth1"])
+
+    def test_get_interfaces_empty_list(self):
+        with mock.patch("moduletests.src.enadiag.os.listdir") as mock_dir:
+            mock_dir.return_value = ["lo"]
+            with contextlib.redirect_stdout(self.output):
+                with self.assertRaises(ValueError) as ex:
+                    moduletests.src.enadiag.get_interfaces()
+                    self.assertEqual(self.output.getvalue(),"[WARN] No interfaces found")
+
+    @mock.patch("moduletests.src.enadiag.os.listdir", side_effect=OSError)
```

Comment:
Should this test be testing for the WARN message and the empty list return value?

Reply:
Added self.assertEqual(len(interfaces), 0

Reply:
Seems this is still missing.

Reply:
This is a cursed PR for me, I think. Somehow thought this was referring to the test above. Resolved now.